### PR TITLE
Add left/right modifier support to KeyBindingContainer

### PR DESCRIPTION
**Pull request by peppy**
Don't really have much to say about this one. Pretty hellish to get the implementation right, and likely not going to be a simple one to review/understand, but it works.

May need to assess whether the array allocation overhead is an issue (although I think the scenarios where it is called are already limited to when input changes, so should be okay).

Closes https://github.com/ppy/osu-framework/issues/2136.

### CODE REVIEW

File: osu.Framework/Input/Bindings/KeyBindingContainer.cs [hunk 1]
```diff
@@ -284,12 +297,14 @@ private void releasePressedActions()
...
         {
+            Logger.Log($"new key released {releasedKey}");
+
             var pressedCombination = KeyCombination.FromInputState(state);
 
             // we don't want to consider exact matching here as we are dealing with bindings, not actions.
             var newlyReleased = pressedBindings.Where(b => !b.KeyCombination.IsPressed(pressedCombination, KeyCombinationMatchingMode.Any)).ToList();
 
-            Trace.Assert(newlyReleased.All(b => b.KeyCombination.Keys.Contains(releasedKey)));
+            // Trace.Assert(newlyReleased.All(b => KeyCombination.ContainsKey(b.KeyCombination.Keys, releasedKey)));
```
Comment (bdach) [inline osu.Framework/Input/Bindings/KeyBindingContainer.cs:307]: Was this purposefully commented out, or just for testing? The log statements in this file are for debugging, I assume, but I'm not sure about this one.

Comment (peppy) [reply osu.Framework/Input/Bindings/KeyBindingContainer.cs:307]: The debug logging and this needs to be reverted, completely forgot.

Hopefully this doesn't trigger...

**Comment by peppy** [comment]
Method complexity CI failure is likely not fixable, safe to ignore.

**Review by smoogipoo — APPROVED** [review]
This looks pretty good to me. The only minor concern I have is the allocation on `FromKeys()`, but I figure both this and the O(N^2)'dness of `ContainsKey()` are rare enough to not cause issues.

**Pull request merged**